sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java

```diff
@@ -198,36 +203,47 @@ public ContentValues nextValues(Cursor cursor) {
      * new one can fit. If the log is larger than the max table size, database will be cleared and
      * the log is not inserted.
      *
-     * @param values The entry to be stored.
+     * @param values         The entry to be stored.
+     * @param priorityColumn When storage full and deleting data, use this column to determine which entries to delete first.
      * @return If a log was inserted, the database identifier. Otherwise -1.
      */
     @SuppressWarnings("TryFinallyCanBeTryWithResources")
-    public long put(@NonNull ContentValues values) {
+    public long put(@NonNull ContentValues values, @NonNull String priorityColumn) {
+        Long id = null;
+        Cursor cursor = null;
         try {
-            while (true) {
+            while (id == null) {
                 try {
 
                     /* Insert data. */
-                    return getDatabase().insertOrThrow(mTable, null, values);
+                    id = getDatabase().insertOrThrow(mTable, null, values);
                 } catch (SQLiteFullException e) {
 
                     /* Delete the oldest log. */
-                    Cursor cursor = getCursor(SQLiteUtils.newSQLiteQueryBuilder(), null, true);
-                    try {
-                        if (cursor.moveToNext()) {
-                            delete(cursor.getLong(0));
-                        } else {
-                            return -1;
-                        }
-                    } finally {
-                        cursor.close();
+                    if (cursor == null) {
+                        String priority = values.getAsString(priorityColumn);
+                        SQLiteQueryBuilder queryBuilder = SQLiteUtils.newSQLiteQueryBuilder();
+                        queryBuilder.appendWhere(priorityColumn + " <= ?");
+                        cursor = getCursor(queryBuilder, SELECT_PRIMARY_KEY, new String[]{priority}, priorityColumn);
```

Comment:
It would be better to explicitly specify sort order for primary key. `priority` then `oid`